### PR TITLE
Apply eclipse code style

### PR DESCRIPTION
**Pull request by snuyanzin**
This is kind another attempt to apply styling.
Finally I refused using google-format-plugin since it's not very configurable...
From that point of view eclipse-formatter gives much more freedom (could be applied in IntellijIdea as well)
Full configuration could be found at _src/main/config/eclipse-java-formatter.xml_

At the same time `spotless` gives autoformatting based on provided configurations

so just run `mvn spotless:apply` or `./mvnw spotless:apply` (the PR is mostly a result of this command)

Current configuration does not touch javadoc => should not be issues like mentioned at https://github.com/datafaker-net/datafaker/pull/220#issuecomment-1159690778

it also should fix https://github.com/datafaker-net/datafaker/issues/469
// cc @NarekDW @bodiam

### CODE REVIEW

**Comment by bodiam** [comment]
I don't mind a reformat here and there, but I am not going to change my IntelliJ defaults or load an external configuration file, nor ask anyone else who contributes to the project to do the same.

File: src/main/java/net/datafaker/Faker.java [hunk 1]
```diff
@@ -13,7 +13,8 @@
 import java.util.Locale;
 import java.util.Random;
 
-public class Faker extends BaseFaker implements BaseProviders, SportProviders, FoodProviders, MovieProviders, VideoGameProviders {
+public class Faker extends BaseFaker
```
Comment (bodiam) [inline src/main/java/net/datafaker/Faker.java:16]: @snuyanzin can we fix this? Maybe set the line length to 200 or so?

File: src/main/java/net/datafaker/formats/Csv.java [hunk 2]
```diff
@@ -54,7 +55,8 @@ public String get() {
         for (int line = 0; line < totalLines; line++) {
             int rowNum = line;
             int current = sb.length();
-            addLine(sb, integer -> columns.get(integer).getValue(res == null || res.isEmpty() ? null : res.get(rowNum)));
+            addLine(sb,
```
Comment (bodiam) [inline src/main/java/net/datafaker/formats/Csv.java:58]: this formatting is pretty nasty

File: src/main/java/net/datafaker/formats/Csv.java [hunk 3]
```diff
@@ -178,23 +180,24 @@ public final CsvCollectionBasedBuilder<T> columns(Function<T, String>... columns
...
+                    .collect(Collectors.toList()).toArray(new Function[0]);
             return this;
         }
 
         public Csv<T> build() {
             if (headers != null && columnValues != null && headers.length != columnValues.length) {
                 throw new IllegalArgumentException("Length of headers should be equal to length of columns");
             }
-            List<CollectionColumn<T>> cols = columnValues == null ? Collections.emptyList() : new ArrayList<>(columnValues.length);
+            List<CollectionColumn<T>> cols = columnValues == null ? Collections.emptyList()
```
Comment (bodiam) [inline src/main/java/net/datafaker/formats/Csv.java:192]: :-(

File: src/main/java/net/datafaker/idnumbers/pt/br/IdNumberGeneratorPtBrUtil.java [hunk 4]
```diff
@@ -31,17 +31,20 @@ public static String cnpj(BaseProviders faker, boolean formatted, boolean valid,
...
 
             int d1 = digit(calculateWeight(cnpj.substring(4, 12), 9) + calculateWeight(cnpj.substring(0, 4), 5));
-            int d2 = digit((d1 * 2) + calculateWeight(cnpj.substring(5, 12), 9) + calculateWeight(cnpj.substring(0, 5), 6));
+            int d2 = digit(
+                    (d1 * 2) + calculateWeight(cnpj.substring(5, 12), 9) + calculateWeight(cnpj.substring(0, 5), 6));
 
             cnpj = (cnpj + d1) + d2;
         } else {
-            cnpj = String.valueOf(faker.random().nextInt(1000000000) + (faker.random().nextInt(90) + 10) * 1000000000000L);
+            cnpj = String
```
Comment (bodiam) [inline src/main/java/net/datafaker/idnumbers/pt/br/IdNumberGeneratorPtBrUtil.java:39]: :-( :-( :-(

File: src/main/java/net/datafaker/service/FakeValuesGrouping.java [hunk 5]
```diff
@@ -31,7 +31,7 @@ public void add(FakeValuesInterface fakeValue) {
     }
 
     @Override
-    @SuppressWarnings({"unchecked", "rawtypes"})
+    @SuppressWarnings({ "unchecked", "rawtypes" })
```
Comment (bodiam) [inline src/main/java/net/datafaker/service/FakeValuesGrouping.java:34]: noooo

File: src/test/java/net/datafaker/providers/base/CompanyTest.java [hunk 6]
```diff
@@ -39,7 +39,8 @@ void testBs() {
 
     @Test
     void testLogo() {
-        assertThat(faker.company().logo()).matches("https://pigment.github.io/fake-logos/logos/medium/color/\\d+\\.png");
+        assertThat(faker.company().logo())
```
Comment (bodiam) [inline src/test/java/net/datafaker/providers/base/CompanyTest.java:42]: noooooooooooooooooooooo

**Comment by NarekDW** [comment]
Then let's engage everybody (somehow) to use Intelij default settings...

File: src/main/java/net/datafaker/service/FakeValuesGrouping.java (hunk 5 above)
Comment (snuyanzin) [reply src/main/java/net/datafaker/service/FakeValuesGrouping.java]: this switch off, now should be ok

File: src/test/java/net/datafaker/providers/base/CompanyTest.java (hunk 6 above)
Comment (snuyanzin) [reply src/test/java/net/datafaker/providers/base/CompanyTest.java]: solved by setting line limit to 200

File: src/main/java/net/datafaker/idnumbers/pt/br/IdNumberGeneratorPtBrUtil.java (hunk 4 above)
Comment (snuyanzin) [reply src/main/java/net/datafaker/idnumbers/pt/br/IdNumberGeneratorPtBrUtil.java]: solved by setting line limit to 200

File: src/main/java/net/datafaker/formats/Csv.java (hunk 3 above)
Comment (snuyanzin) [reply src/main/java/net/datafaker/formats/Csv.java]: solved by setting line limit to 200

File: src/main/java/net/datafaker/formats/Csv.java (hunk 2 above)
Comment (snuyanzin) [reply src/main/java/net/datafaker/formats/Csv.java]: solved by setting line limit to 200

File: src/main/java/net/datafaker/Faker.java (hunk 1 above)
Comment (snuyanzin) [reply src/main/java/net/datafaker/Faker.java]: solved by setting line limit to 200

File: src/test/java/net/datafaker/formats/XmlTest.java [hunk 7]
```diff
@@ -21,18 +21,15 @@ void xmlPrettyTest(Xml.XmlNode xmlNode, String expected) {
...
-            Arguments.of(new Xml.XmlNode("root",
-                    map(entry("attribute1", "value1"), entry("attribute2", "value2")), "value"),
-                "<root attribute1=\"value1\" attribute2=\"value2\">value</root>"),
-            Arguments.of(new Xml.XmlNode("root",
-                    map(entry("attribute1", "value1"), entry("attribute2", "value2")),
-                    Collections.singletonList(new Xml.XmlNode("child", "value"))),
-                "<root attribute1=\"value1\" attribute2=\"value2\">" + System.lineSeparator() + "    <child>value</child>" + System.lineSeparator() + "</root>"),
-            Arguments.of(new Xml.XmlNode("root", "<> value\""), "<root>&lt;&gt; value&quot;</root>")
-        );
+        return Stream.of(Arguments.of(new Xml.XmlNode("root", Collections.emptyList()), "<root/>"), Arguments.of(new Xml.XmlNode("root", "value"), "<root>value</root>"), Arguments.of(new Xml.XmlNode(
```
Comment (bodiam) [inline src/test/java/net/datafaker/formats/XmlTest.java:24]: Any chance to get this into a somewhat more reasonable state? I think the previous formatting was quite a bit better?

**Comment by bodiam** [comment]
@snuyanzin How do you configure the formatting?

**Comment by snuyanzin** [comment]
configuration is in this file _src/main/config/eclipse-java-formatter.xml_

File: src/test/java/net/datafaker/idnumbers/PeselNumberTest.java [hunk 8]
```diff
@@ -140,22 +140,22 @@ private void assertBirthDate(LocalDate givenBirthDate, String gotPesel) {
...
-            case FEMALE:
-                assertThat(gotGenderDigit % 2).isZero();
-                break;
-            case MALE:
-                assertThat(gotGenderDigit % 2).isOne();
-                break;
-            default:
-            case ANY:
-                break;
+        case FEMALE:
```
[the remark this answers is not in the document]
Comment (snuyanzin) [reply src/test/java/net/datafaker/idnumbers/PeselNumberTest.java]: yes, this is now done

File: src/test/java/net/datafaker/formats/XmlTest.java (hunk 7 above)
Comment (snuyanzin) [reply src/test/java/net/datafaker/formats/XmlTest.java]: should be better now

**Comment by snuyanzin** [comment]
i don't use it every day however have it installed

**Comment by RockyMM** [comment]
Have you guys ever tried https://github.com/palantir/palantir-java-format It runs with spotless out of the box, and produces *much* more sensible formatting, at least much closer to my liking.

However, in big projects the code style is usually enforced, either by pre-commit hooks, or by CI actions or by using both. I don't like everything about it, but it saves quite a lot of hassle when merging or rebasing.

**Comment by snuyanzin** [comment]
Frankly speaking no
Are you familiar with the way of configuration of it or where to get this info?
the main task is to adapt configuration to current code base

**Comment by RockyMM** [comment]
palantir-java-format is a fork of google-java-format. Both formatters are **strongly** opinionated. In other words, you cannot configure them 😅 

However, the main motivation of the google-java-format is the code review on laptop screens. The main motivation of palantir-java-format is readability in the IDE, whit code review coming second. This is why google is applying 80 character limit, and palantir is applying 120 character limit.

Palantir also recognizes streams, factory methods and formats them ... well, very closely how I used to do before it.

Maybe you could try it out (I could provide a quick fork or PR) and see if you like it.

**Comment by bodiam** [comment]
Closing this, cause well, discussions about 80, 120 or 160 characters are not going to benefit this project substantially.

**Pull request closed without merging**